datalad/distribution/get.py

```diff
@@ -49,53 +51,209 @@
+            results.append(cur_ds)
+            if not fulfill_datasets:
+                lgr.debug(
+                    "Will not get any content in subdataset %s without recursion enabled",
+                    cur_ds)
+                yield results
+                continue
+
+        # needs to be an annex:
+        found_an_annex = isinstance(cur_ds.repo, AnnexRepo)
+        if not found_an_annex:
+            lgr.debug("Found no annex at %s. Skipped.", cur_ds)
+            if results:
+                yield results
+            continue
+        lgr.info("Getting %s file/dir(s) of dataset %s ...",
+                 len(content), cur_ds)
+
+        results.extend(cur_ds.repo.get(
+            content,
+            options=['--from=%s' % source] if source else [],
+            jobs=jobs))
+
+        if refpath:
+            # adapt relative paths reported by annex to be relative some
+            # reference
+            if ds_path != refpath:
+                for lr in results:
+                    if isinstance(lr, dict):
+                        lr['file'] = relpath(opj(ds_path, lr['file']), refpath)
+        yield results
+
+
+def _recursive_install_subds_underneath(ds, recursion_limit, start=None):
+    from .install import _install_subds_from_flexible_source
+    content_by_ds = {}
+    if recursion_limit is not None and recursion_limit <= 0:
+        return content_by_ds
+    # loop over submodules not subdatasets to get the url right away
+    # install using helper that give some flexibility regarding where to
+    # get the module from
+    for sub in ds.repo.get_submodules():
+        subds = Dataset(opj(ds.path, sub.path))
+        if start is not None and not subds.path.startswith(_with_sep(start)):
+            # this one we can ignore, not underneath the start path
+            continue
+        if not subds.is_installed():
+            try:
+                subds = _install_subds_from_flexible_source(
+                    ds, sub.path, sub.url)
+                # we want the entire thing, but mark this subdataset
+                # as automatically installed
+                content_by_ds[subds.path] = [curdir]
+            except Exception as e:
+                # skip, if we didn't manage to install subdataset
+                lgr.warning(
+                    "Installation of subdatasets %s failed, skipped", subds)
+                lgr.debug("Installation attempt failed with exception: %s",
+                          exc_str(e))
+                continue
+            # otherwise recurse
+            # we can skip the start expression, we know we are within
+            content_by_ds.update(_recursive_install_subds_underneath(
+                subds,
+                recursion_limit=None if recursion_limit is None else recursion_limit - 1
+            ))
+    return content_by_ds
+
+
+def _unwind(generators):
+    res = []
+    for item in generators:
+        res.extend(item)
+    return res
```

Comment:
as far as I see, you just wanted to say `list(generator)`, no need in custom `_unwind`